pkg/controller/datadogagentdeployment/poddisruptionbudget.go

```diff
@@ -0,0 +1,167 @@
+	newPdb := builder(dad)
+	// Set DatadogAgentDeployment instance  instance as the owner and controller
+	if err := controllerutil.SetControllerReference(dad, newPdb, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := r.client.Create(context.TODO(), newPdb); err != nil {
+		return reconcile.Result{}, err
+	}
+	logger.Info("Create PDB", "name", newPdb.Name)
+	r.recorder.Event(dad, corev1.EventTypeNormal, "Create PDB", fmt.Sprintf("%s/%s", newPdb.Namespace, newPdb.Name))
+
+	return reconcile.Result{Requeue: true}, nil
+}
+
+func (r *ReconcileDatadogAgentDeployment) updateIfNeededPDB(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment, currentPDB *policyv1.PodDisruptionBudget, newStatus *datadoghqv1alpha1.DatadogAgentDeploymentStatus, builder pdbBuilder) (reconcile.Result, error) {
+	if !ownedByDatadogOperator(currentPDB.OwnerReferences) {
+		return reconcile.Result{}, nil
+	}
+	newPDB := builder(dad)
+	result := reconcile.Result{}
+	if !(apiequality.Semantic.DeepEqual(newPDB.Spec, currentPDB.Spec) &&
+		apiequality.Semantic.DeepEqual(newPDB.Labels, currentPDB.Labels) &&
+		apiequality.Semantic.DeepEqual(newPDB.Annotations, currentPDB.Annotations)) {
+
+		updatedPDB := currentPDB.DeepCopy()
+		updatedPDB.Labels = newPDB.Labels
+		updatedPDB.Annotations = newPDB.Annotations
+		updatedPDB.Spec = newPDB.Spec
+
+		if err := r.client.Update(context.TODO(), updatedPDB); err != nil {
+			return reconcile.Result{}, err
+		}
+		r.recorder.Event(dad, corev1.EventTypeNormal, "Update PDB", fmt.Sprintf("%s/%s", updatedPDB.Namespace, updatedPDB.Name))
+		result.Requeue = true
+	}
+
+	return result, nil
+}
+
+func (r *ReconcileDatadogAgentDeployment) cleanupPDB(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment, newStatus *datadoghqv1alpha1.DatadogAgentDeploymentStatus, pdbName string) (reconcile.Result, error) {
+	// checks token secret
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: dad.Namespace, Name: pdbName}, pdb)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	if ownedByDatadogOperator(pdb.OwnerReferences) {
+		err = r.client.Delete(context.TODO(), pdb)
+	}
+
+	return reconcile.Result{}, err
+}
+
+func buildClusterAgentPDB(dad *datadoghqv1alpha1.DatadogAgentDeployment) *policyv1.PodDisruptionBudget {
+	labels := getDefaultLabels(dad, datadoghqv1alpha1.DefaultClusterAgentResourceSuffix, getClusterAgentVersion(dad))
+	annotations := getDefaultAnnotations(dad)
+	metadata := metav1.ObjectMeta{
+		Name:        getClusterAgentPDBName(dad),
+		Namespace:   dad.Namespace,
+		Labels:      labels,
+		Annotations: annotations,
+	}
+	matchLabels := map[string]string{
+		datadoghqv1alpha1.AgentDeploymentNameLabelKey:      dad.Name,
+		datadoghqv1alpha1.AgentDeploymentComponentLabelKey: "cluster-agent",
```

Comment:
shouldn't we as `datadoghqv1alpha1.DefaultClusterAgentResourceSuffix` here as well

Reply:
Yep, fixing source of my copy/paste too :D.

Reply:
thanks :D